contracts/utils/ReentrancyGuard.sol

```diff
@@ -62,26 +62,37 @@ abstract contract ReentrancyGuard {
     }
 
     function _nonReentrantBefore() private {
-        // On the first call to nonReentrant, _status will be NOT_ENTERED
-        if (_status == ENTERED) {
-            revert ReentrancyGuardReentrantCall();
+        assembly {
+            // Load the value of _status
+            let status := sload(_status.slot)
+            // If status is ENTERED, throw an error
+            if eq(status, 2) {
+                mstore(0x00, 0x4e487b71) // keccak256("Error(string)")
+                mstore(0x20, 0x20) // Data length (32 bytes)
+                revert(0x00, 0x24) // Revert with (0x00 to 0x24)
+            }
+            // Set _status to ENTERED
+            sstore(_status.slot, 2)
```

Comment:
in general, we should be using `ENTERED` and `NOT_ENTERED` constants instead of the defined value.